### ISSUE TITLE
Username menu in admin interface does not have standard controls

### ISSUE THREAD

**Issue by keelanfh** [localgovdrupal/localgov#568]
### Steps to replicate
* Tab to User Name link, tab to elements in the list.

![image](https://github.com/localgovdrupal/localgov/assets/19519457/98fd3160-bfad-42dd-a0da-6996f99e2dd3)

### Actual results

* What appears to be a menu, has been programmed as a list only. Standard keys for menus do not work with the list. You can’t close the list while on it, Esc does not close the menu
* Although this appears to be a menu it has not been programmed using WAI-ARIA principles which would ensure it was accessible and be useable with keys for menus.

### Expected results

* The menu should be dismissible when it is opened by the Esc key.
* Aria principles menus are applied so that the menu can be accessed by keyboard only users and screen reader users and would meet the WCAG 2.1 guidelines.

### WCAG 2.1 references
2.1.1. All functionality is operable with a keyboard.
https://www.w3.org/WAI/WCAG21/Understanding/keyboard.html
4.1.2 Use standard controls, or ensure that non-standard controls are properly marked up using WAI-ARIA
https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html
